### PR TITLE
Add clippy, lints and nightly for travis.

### PR DESCRIPTION
**Pull request by ghost**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/socket_addr/2)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
Thanks for the pull request, and welcome! The MaidSafe team is excited to review your changes, and you should hear from @canndrew (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. The way GitHub handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [CONTRIBUTOR.md](../blob/master/CONTRIBUTOR) for more information.

File: src/lib.rs [hunk 1]
```diff
@@ -15,9 +15,32 @@
...
-/// There is no intent here to provide many more traits (perhaps serde).
-/// Deref is implemented to allow easy access to the underlying `net::ScoketAddr`
+//! A very simple ScoketAddr structure that is serialisable and makes use of the `ip` crate.
+//! There is no intent here to provide many more traits (perhaps serde).
+//! Deref is implemented to allow easy access to the underlying `net::ScoketAddr`
+
+#![doc(html_logo_url =
+           "https://raw.githubusercontent.com/maidsafe/QA/master/Images/maidsafe_logo.png",
+       html_favicon_url = "http://maidsafe.net/img/favicon.ico",
+       html_root_url = "http://maidsafe.github.io/secure_serialisation")]
```
Comment (Fraser999) [inline src/lib.rs]: The URL is to the wrong project here.

**Comment by ghost** [comment]
Nice catch, I've updated it.

**Comment by Fraser999** [comment]
Thanks.

**Comment by Fraser999** [comment]
Reviewed 2 of 3 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/socket_addr/2)_

**Pull request merged**